integration-tests/environments/react-native/metro.config.js

```diff
@@ -16,40 +16,96 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
+// Metro bundler does not support symlinks (see https://github.com/facebook/metro/issues/1).
+// In order to link to our local version of Realm JS in a way which Metro can follow,
+// we need to instead add the module to `resolver.blockList`, and then in `resolver.extraNodeModules`,
+// override the locations for any symlinked modules by specifying a mapping from their name
+// to their actual location on disk.
+
+// In order to achieve this, we recursively check if each file/directory in `node_modules`
+// is a symlink or not. If it is, and if it is specified in our main `package.json` (i.e.
+// it is a direct dependency of our app), we add it to the `blockList` and
+// add a mapping for it. We also add it to the list of watched folders, so
+// that any changes to it are picked up by hot reloading.
+
 const path = require("path");
 const fs = require("fs");
 const exclusionList = require("metro-config/src/defaults/exclusionList");
 
-// Read the localDependencies property of package.json and resolve the relative path values
-const appPackageJson = readJson(__dirname, "package.json");
-const localDependencies = Object.fromEntries(
-  Object.entries(appPackageJson.localDependencies || {}).map(([localName, relativePath]) => [
-    localName,
-    path.resolve(__dirname, relativePath),
-  ]),
-);
-
 function readJson(...pathSegemnts) {
   const filePath = path.join(...pathSegemnts);
   return JSON.parse(fs.readFileSync(filePath, "utf8"));
 }
 
-const watchFolders = Object.values(localDependencies);
+function getLinkedDependencies(packagePath, recursive = true, exclude = new Set()) {
```

Comment:
Do we need the `recursion` flag?  As this is not a utility function being used for multiple things, I think having support for `recursion` just complicates the code.

Reply:
The default is actually to recurse (as that's needed for our purpose). For example Realm JS has a symbolically linked dependency on `realm-network-transport`, which we also want to resolve. I do agree that since we're always recursing, having that as an argument is overkill (although it was useful when debugging).

Reply:
Ah, I overlooked the default value.  I suppose if it has some use, we can keep it in.